src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/ViewProviders/GraphViewProviderBase.cs

```diff
@@ -121,7 +122,7 @@ IEnumerable<string> GetExistingChildren()
         public virtual bool MatchSearchResults(
             IDependency topLevelDependency,
             Dictionary<string, HashSet<IDependency>> searchResultsPerContext,
-            out HashSet<IDependency>? topLevelDependencyMatches)
+            [MaybeNullWhen(returnValue: false)] out HashSet<IDependency>? topLevelDependencyMatches)
```

Comment:
Does this mean that the out param definitely won't be null when the method returns true?

Reply:
The above construct is incorrect. Pushed a correction.

The four attributes (AllowNull/DisallowNull/NotNull/MaybeNull) and the conditional form above all override the type's declared nullability. So in the above example the HashSet should be non-null with the attribute specifying a null return when false (or a nullable type with NotNullWhen(true)).

Reply:
Yep, the update makes much more sense. Couldn't work out what "maybe null" meant when the parameter type was already nullable.